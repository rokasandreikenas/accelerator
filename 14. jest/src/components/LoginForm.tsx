import { useState } from "react";

interface LoginFormProps {
  onSubmit: (username: string, password: string) => void;
}

const LoginForm: React.FC<LoginFormProps> = ({ onSubmit }) => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [errors, setErrors] = useState<{
    username?: string;
    password?: string;
  }>({});

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();

    const validationErrors: { username?: string; password?: string } = {};
    if (!username) validationErrors.username = "Username is required";
    if (!password) validationErrors.password = "Password is required";

    if (Object.keys(validationErrors).length > 0) {
      setErrors(validationErrors);
    } else {
      onSubmit(username, password);
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <label>
        Username:
        <input
          type="text"
          value={username}
          onChange={(e) => setUsername(e.target.value)}
        />
        {errors.username && <div>{errors.username}</div>}
      </label>
      <label>
        Password:
        <input
          type="password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
        {errors.password && <div>{errors.password}</div>}
      </label>
      <button type="submit">Login</button>
    </form>
  );
};

export default LoginForm;
