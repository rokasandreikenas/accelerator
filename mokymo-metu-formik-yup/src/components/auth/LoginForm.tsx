import { Formik, Form } from "formik";
import { loginFormInitialValues } from "./consts";
import { LoginFormValues } from "./types";
import { loginUser } from "./api";
import LoginFormFields from "./LoginFormFields";

const LoginForm = () => {
  const handleSubmit = async (values: LoginFormValues) => {
    try {
      await loginUser(values);
      alert("Successfull login");
    } catch (error) {
      console.error(error);
    }
  };

  const validate = (values: LoginFormValues) => {
    const errors: Record<string, string> = {};

    if (!values.email.includes("@")) {
      errors.email = "El. pašte nėra simbolio @";
    }

    if (!values.email) {
      errors.email = "Įveskite reikšmę";
    }

    if (values.password.length < 5) {
      errors.password = "Jūsų slaptažodis yra per trumpas. Min 5 simboliai";
    }
    console.log(errors);
    return errors;
  };

  // {password: "Erorras"}
  return (
    <div>
      <Formik
        initialValues={loginFormInitialValues}
        onSubmit={handleSubmit}
        validate={validate}
      >
        {/* {({ values, errors }) => (
          <Form>
            <h2>{values.email}</h2>
            <Field placeholder="El. pastas" name="email" />
            {errors.email && <div style={{ color: "red" }}>{errors.email}</div>}
            <br />
            <Field placeholder="Slaptazodis" type="password" name="password" />
            {errors.password && (
              <div style={{ color: "red" }}>{errors.password}</div>
            )}
            <br />
            <button type="submit">Login</button>
          </Form>
        )} */}
        <Form>
          <LoginFormFields />
          <button type="submit">Login</button>
        </Form>
      </Formik>
    </div>
  );
};

export default LoginForm;
