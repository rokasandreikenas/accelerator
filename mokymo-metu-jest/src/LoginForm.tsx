import { useState } from "react";
import Heading from "./Heading";
import LoginFormInputs from "./LoginFormInputs";

interface LoginProps {
  onSubmit: (email: string, password: string) => void;
}

const LoginForm = ({ onSubmit }: LoginProps) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  return (
    <form
      onSubmit={(event) => {
        event.preventDefault();
        onSubmit(email, password);
      }}
    >
      <Heading>LoginForm</Heading>
      <LoginFormInputs
        email={email}
        password={password}
        setEmail={setEmail}
        setPassword={setPassword}
      />
      <button type="submit">Login</button>
    </form>
  );
};

export default LoginForm;
