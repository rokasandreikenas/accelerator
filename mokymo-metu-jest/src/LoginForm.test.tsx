import { render, screen, fireEvent } from "@testing-library/react";
import LoginForm from "./LoginForm";

describe("<LoginForm />", () => {
  const onSubmit = jest.fn();

  test("renders component successfuly", () => {
    render(<LoginForm onSubmit={onSubmit} />);
  });

  test("submits form data", async () => {
    render(<LoginForm onSubmit={onSubmit} />);

    const emailInput = screen.getByRole("textbox", { name: "Email" });
    const passwordInput = screen.getByPlaceholderText("password");

    fireEvent.change(emailInput, { target: { value: "rokas@gmail.com" } });
    fireEvent.change(passwordInput, { target: { value: "rokas123" } });

    const loginButton = screen.getByRole("button", { name: "Login" });

    fireEvent.click(loginButton);

    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(onSubmit).toHaveBeenCalledWith("rokas@gmail.com", "rokas123");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<LoginForm onSubmit={onSubmit} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
