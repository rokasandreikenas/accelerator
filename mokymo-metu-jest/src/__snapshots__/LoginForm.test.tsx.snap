// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <h2>
      LoginForm
    </h2>
    <input
      aria-label="Email"
      name="email"
      placeholder="email"
      value=""
    />
    <input
      aria-label="Password"
      name="password"
      placeholder="password"
      value=""
    />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;
